jerry-libm/jerry-libm-internal.h

```diff
@@ -38,6 +39,7 @@
      defined (__xtensa__))
```

Comment:
In case of GCC/Clang we could use the compiler's `__BYTE_ORDER__` macro to check if it is equal to `__ORDER_LITTLE_ENDIAN__` and then defined the `__LITTLE_ENDIAN` macro here. That way we don't need to add extra defines to the toolchain/build files. Would that solve this problem?

Reply:
If it doesn't make problem for other architectures, it is simpler option.
I should work for mips. I will test it.

Reply:
or I think it is also possible to simply check the `MIPSEL` or `_MIPSEL` macro definitions.

Reply:
True, `MIPSEL` macros are defined and would solve this specific problem but maybe generic option would be a good idea (in case of new architectures)? Is there reason each target is checked separately instead of checking endians?

Reply:
This libm is from the https://www.netlib.org/fdlibm/ library and the macros were used that way. At that time (of import) we did not changed the architecture checks. However, I'm fine with changing this to use the byte order macro.

Reply:
ahh.. I can think one problem: if someone uses an older compiler which does not support the byte order macro, I think pre gcc 4.6 is one such compiler.

Reply:
Maybe the best way would be to check the MIPSEL macro if it is defined, and also check if the byte order macro is defined and set to little endian mode. That way all sides are covered.

Reply:
Updated so that if there is `__BYTE_ORDER__` endians are checked there and if not gets to checking targets with `MIPSEL` added.